api/app/Notifications/System.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+namespace App\Notifications;
+
+use App\Enums\Language;
+use App\Models\User;
+use App\Notifications\Messages\GcNotifyEmailMessage;
+use Illuminate\Bus\Queueable;
+use Illuminate\Notifications\Notification;
+use Illuminate\Support\Facades\View;
+
+class System extends Notification implements CanBeSentViaGcNotifyEmail
+{
+    use Queueable;
+
+    /**
+     * Create a new notification instance.
+     */
+    public function __construct(
+        public string $emailSubjectEn,
+        public string $emailSubjectFr,
+        public string $emailContentEn,
+        public string $emailContentFr,
+        public string $inAppMessageEn,
+        public string $inAppMessageFr,
+        public string $inAppHrefEn,
+        public string $inAppHrefFr,
+    ) {
+    }
+
+    /**
+     * Get the notification's delivery channels.
+     *
+     * @return array<int, string>
+     */
+    public function via(User $notifiable): array
+    {
+        // $notificationFamily = NotificationFamily::SYSTEM_MESSAGE->name;
+        // Can't ignore system messages
+        return ['database', GcNotifyEmailChannel::class];
+    }
+
+    /**
+     * Get the array representation of the notification.
+     * https://laravel.com/docs/10.x/notifications#formatting-database-notifications
+     *
+     * @return array<string, mixed>
+     */
+    public function toArray(object $notifiable): array
+    {
+        return [
+            'message' => [
+                'en' => View::make($this->inAppMessageEn, ['user' => $notifiable])->render(),
+                'fr' => View::make($this->inAppMessageFr, ['user' => $notifiable])->render(),
+            ],
+            'href' => [
+                'en' => trim(View::make($this->inAppHrefEn, ['user' => $notifiable])->render()),
+                'fr' => trim(View::make($this->inAppHrefFr, ['user' => $notifiable])->render()),
+            ],
+        ];
+    }
+
+    /**
+     * Get the GC Notify representation of the notification.
+     */
+    public function toGcNotifyEmail(User $notifiable): GcNotifyEmailMessage
+    {
+        $locale = $this->locale ?? $notifiable->preferredLocale();
+
+        if ($locale == Language::EN->value) {
```

Comment:
Bilingual notification could have been easier in this case. No need to check every user's preferred locale.

Reply:
Yeah, I remember talking about bilingual notifications for the talent search request notification.  We did a bilingual one there because the user wasn't logged in but there was a strong preference for localized notifications when possible.